### PR TITLE
refactor: make the prepare process clearer and modify the resolve alias

### PR DESCRIPTION
**Pull request by meteorlxy**
- Write notfound route to the temp route file instead of adding in `app.js`
- Make the `resolveOptions` process in `prepare` clearer
- Change the `@theme` alias to theme directory. Add `@themeLayout` and `@themeNotFound` aliases for `Layout.vue` and `NotFound.vue` respectively.

### CODE REVIEW

File: lib/prepare.js [hunk 1]
```diff
@@ -279,30 +292,38 @@ async function genRoutesFile ({ siteData: { pages }, sourceDir, pageFiles }) {
...
     if (/\/$/.test(pagePath)) {
-      code += `,{
-        path: ${JSON.stringify(pagePath + 'index.html')},
-        redirect: ${JSON.stringify(pagePath)}
-      }`
+      code += `,
+  {
+    path: ${JSON.stringify(pagePath + 'index.html')},
+    redirect: ${JSON.stringify(pagePath)}
+  }`
```
Comment (ulivz) [inline lib/prepare.js]: This indent looks not very good.

**Comment by meteorlxy** [comment]
@yyx990803 @ulivz 

Review wanted~~~

**Comment by ulivz** [comment]
@meteorlxy I need to take some time to run your code at my area, please wait for some time.

**Comment by meteorlxy** [comment]
@ulivz Hope to be merged ASAP~ I want to fix some issue or add some features based on this PR.

File: lib/prepare.js [hunk 2]
```diff
@@ -111,6 +105,11 @@ async function resolveOptions (sourceDir) {
     })
   }
 
+  // resolve outDir
+  const outDir = siteConfig.dest
+    ? path.resolve(siteConfig.dest)
+    : path.resolve(sourceDir, '.vuepress/dist')
+
   // resolve theme
```
Comment (yyx990803) [inline lib/prepare.js]: move this section down? There are currently two `// resolve theme` blocks.

**Comment by yyx990803** [comment]
LGTM after the small change requested.

**Comment by meteorlxy** [comment]
Tweaked, sir!

**Comment by meteorlxy** [comment]
@yyx990803 Tweaked, sir!

**Comment by ulivz** [comment]
@meteorlxy Do you use slack or telegram? I can create a channel for that.

**Comment by meteorlxy** [comment]
Oh, I didn't use them before, but I can try them. Which one do you prefer?

(How about Wechat, LOL)

**Comment by meteorlxy** [comment]
@ulivz I tried slack, and my email is meteorlxy.cn@gmail.com

**Comment by ulivz** [comment]
@meteorlxy I have invited you to `vuepress.slack.com`.

**Comment by ycmjason** [comment]
@ulivz sorry but I just saw this. Any chance I could join the slack or is this selected personell only? 😂😂😂 Wouldn't mind at all if not possible. 😁

**Comment by ulivz** [comment]
@ycmjason sure, can you provide the email？Let me invite you.

**Comment by ycmjason** [comment]
@ulivz me@ycmjason.com
Wohoo!

**Comment by hmatalonga** [comment]
@ulivz 

Could I also join the vuepress slack? It doesn't hurt to try asking :joy: 

I would like to follow along the Plugin API plans and future discussions...

**Pull request merged**